exporters/secret/exporter.go

```diff
@@ -0,0 +1,169 @@
+package secret
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/giantswarm/k8sclient/v5/pkg/k8srestconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	fieldSelector = "type=kubernetes.io/tls"
+	certType      = "secret"
+)
+
+type Config struct {
+	Namespaces []string
+}
+
+type Exporter struct {
+	cert      *prometheus.Desc
+	ctx       context.Context
+	k8sClient *kubernetes.Clientset
+	logger    micrologger.Logger
+
+	namespaces []string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Namespaces: []string{},
+	}
+}
+
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	e.logger.Log("info", "start collecting metrics")
+
+	var clusterSecrets []v1.Secret
+
+	listOpts := metav1.ListOptions{
+		FieldSelector: fieldSelector,
+	}
+
+	// Build list of secrets to check
+	if len(e.namespaces) == 0 {
+		// If no namespaces are provided then get all secrets matching selector
+		secrets, err := e.k8sClient.CoreV1().Secrets("").List(e.ctx, listOpts)
```

Comment:
you can deduplicate this and the following block by setting `namespace` equal to `""` if there are zero namespaces